architecture/src/iosMain/kotlin/observable/SimpleDisposable.kt

```diff
@@ -25,22 +25,3 @@ actual class SimpleDisposable actual constructor(onDispose: DisposeHandler) : Ba
         GCScheduler.schedule()
```

Comment:
Incidentally this is now the last remaining reason to have an `expect` and `BaseSimpleDisposable`. Is it still needed with the new GC?

Reply:
I think so. Garbage Collect needs to run immediately after unsubscribing, otherwise Swift code tends to go buggy. If we dont force it, who knows when it runs.

Reply:
I thought it was because in the old code you'd actually get an out of memory error. But there is a completely new GC now, that might detect the actual memory pressure and collect.

It's certainly something we can test at some later point, as this PR is just about concurrency.